code/modules/surgery/organs/external/tails.dm

```diff
@@ -32,6 +34,42 @@
 		else if(type in receiver.dna.species.external_organs)
 			receiver.add_mood_event("wrong_tail_regained", /datum/mood_event/tail_regained_wrong)
 
+/obj/item/organ/external/tail/on_bodypart_insert(obj/item/bodypart/bodypart)
+	var/obj/item/organ/external/spines/our_spines = bodypart.owner.get_organ_slot(ORGAN_SLOT_EXTERNAL_SPINES)
+	if(our_spines)
+		try_insert_tail_spines(bodypart)
+	return ..()
+
+/obj/item/organ/external/tail/on_bodypart_remove(obj/item/bodypart/bodypart)
+	remove_tail_spines(bodypart)
+	return ..()
+
+/// If the owner has spines and an appropriate overlay exists, add a tail spines overlay.
+/obj/item/organ/external/tail/proc/try_insert_tail_spines(obj/item/bodypart/bodypart)
+	// Don't insert another overlay if there already is one.
+	if(tail_spines_overlay)
+		return
+	// If this tail doesn't have a valid set of tail spines, don't insert them
+	var/datum/sprite_accessory/tails/tail_sprite_datum = bodypart_overlay.sprite_datum
+	if(!istype(tail_sprite_datum))
+		return
+	var/tail_spine_key = tail_sprite_datum.spine_key
+	if(!tail_spine_key)
+		return
+
+	tail_spines_overlay = new
+	tail_spines_overlay.tail_spine_key = tail_spine_key
+	var/feature_name = bodypart.owner.dna.features["spines"] //tail spines don't live in DNA, but share feature names with regular spines
+	tail_spines_overlay.set_appearance_from_name(feature_name)
+	bodypart.add_bodypart_overlay(tail_spines_overlay)
+
+/// If we have a tail spines overlay, delete it
+/obj/item/organ/external/tail/proc/remove_tail_spines(obj/item/bodypart/bodypart)
+	if(!tail_spines_overlay)
+		return
+	bodypart.remove_bodypart_overlay(tail_spines_overlay)
+	qdel(tail_spines_overlay)
```

Comment:
This is causing harddels

use 
```suggestion
	QDEL_NULL(tail_spines_overlay)
```
(Might be called something slightly different